File: 204-Count-Primes/204-Count-Primes.m
#import <Foundation/Foundation.h>

@interface NSNumber (CountPrimes)

- (NSUInteger)countPrimes;

@end

@implementation NSNumber (CountPrimes)

- (NSUInteger)countPrimes {
    NSUInteger count = 0;
    NSInteger num = [self integerValue];
    
    BOOL *isPrime = (BOOL*)malloc(num * sizeof(BOOL));
    for (NSUInteger i = 0;i < num; i++) {
        isPrime[i] = YES;
    }
    
    for (NSUInteger i = 2;i < num; i++) {
        if (isPrime[i]) {
            count++;
            
            for (NSUInteger j = 2;i * j < num; j++) {
                isPrime[i * j] = NO;
            }
        }
    }
    
    return count;
}

@end

int main(int argc, char * argv[]) {
    NSLog(@"%ld", [@(10) countPrimes]);
}

File: 204-Count-Primes/204-Count-Primes.swift
class Solution {    
    func countPrimes(_ n: Int) -> Int {
        var isPrime = Array(repeating: true, count: n + 1)
        var count = 0
        
        for num in stride(from: 2, to: n, by: 1) {
            if isPrime[num] {
                count += 1
                
                var i = 2
                var composite = i * num
                while composite < n {
                    isPrime[composite] = false
                    
                    i += 1
                    composite = i * num
                }
            }
        }
        
        return count
    }
}
